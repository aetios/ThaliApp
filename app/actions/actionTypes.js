export const NAVIGATE = 'NAVIGATE';
export const BACK = 'BACK';
export const OPENDRAWER = 'OPENDRAWER';
export const CALENDARRETREIVED = 'CALENDARRETREIVED';
export const CALENDARERROR = 'CALENDARERROR';
export const RESETLOGINSTATE = 'RESETLOGINSTATE';
export const WELCOME = 'WELCOME';
