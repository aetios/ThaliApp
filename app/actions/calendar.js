/**
 * Created by aucke on 3/8/17.
 */
