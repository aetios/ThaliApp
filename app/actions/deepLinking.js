export const DEEPLINK = 'DEEPLINKING_DEEPLINK';

export function deepLink(url, stayInApp = true) {
  return {
    type: DEEPLINK,
    payload: { url, stayInApp },
  };
}
