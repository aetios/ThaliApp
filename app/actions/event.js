export const EVENT = 'EVENT_EVENT';
export const FETCHING = 'EVENT_FETCHING';
export const SUCCESS = 'EVENT_SUCCESS';
export const FAILURE = 'EVENT_FAILURE';

export function event(pk, token) {
  return {
    type: EVENT,
    payload: { pk, token },
  };
}

export function success(eventData, eventRegistrations) {
  return {
    type: SUCCESS,
    payload: { eventData, eventRegistrations },
  };
}

export function fetching() {
  return {
    type: FETCHING,
  };
}

export function failure() {
  return {
    type: FAILURE,
  };
}
