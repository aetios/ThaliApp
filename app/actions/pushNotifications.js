export const REGISTER = 'PUSH_NOTIFICATIONS_REGISTER';
export const INVALIDATE = 'PUSH_NOTIFICATIONS_INVALIDATE';

export function register(categories = null) {
  return { type: REGISTER, categories };
}

export function invalidate() {
  return { type: INVALIDATE };
}
