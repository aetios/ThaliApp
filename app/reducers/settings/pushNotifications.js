import { notificationsSettingsActions as actions } from '../../actions/settings';

export const initialState = {
  categoryList: [],
  status: 'loading',
};

export default function pushNotifications(state = initialState, action = {}) {
  switch (action.type) {
    case actions.SUCCESS:
      return {
        categoryList: action.categoryList,
        status: 'success',
      };
    case actions.FAILURE:
      return {
        ...state,
        status: 'failure',
      };
    default:
      return state;
  }
}
