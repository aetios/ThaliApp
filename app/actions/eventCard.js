import * as types from "./actionTypes";
export function openEvent(eventId) {
  console.log("opening event");
  return {
  };
}
