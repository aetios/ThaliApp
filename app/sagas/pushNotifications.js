import { call, takeEvery, select } from 'redux-saga/effects';
import { Platform } from 'react-native';
import FCM from 'react-native-fcm';

import { apiRequest, tokenSelector } from '../url';
import * as pushNotificationsActions from '../actions/pushNotifications';

const register = function* register() {
  const token = yield select(tokenSelector);

  if (token === undefined) {
    // There is no token, thus do nothing
    return;
  }

  let pushToken;
  if (Platform.OS === 'ios') {
    try {
      // this throws an error when the permissions are denied
      yield call(FCM.requestPermissions);
      pushToken = yield call(FCM.getFCMToken);
    } catch (err) {
      // return and do nothing since we have no token
      return;
    }
  } else {
    pushToken = yield call(FCM.getFCMToken);
  }

  const data = {
    method: 'POST',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
      Authorization: `Token ${token}`,
    },
    body: JSON.stringify({
      registration_id: pushToken,
      type: Platform.OS,
    }),
  };

  try {
    yield call(apiRequest, 'devices', data);
  } catch (err) {
    // eat error, om nom nom
  }
};

const invalidate = function* invalidate() {
  yield call(FCM.deleteInstanceId);
};

const pushNotificationsSaga = function* pushNotificationsSaga() {
  yield takeEvery(pushNotificationsActions.REGISTER, register);
  yield takeEvery(pushNotificationsActions.INVALIDATE, invalidate);
};

export default pushNotificationsSaga;
