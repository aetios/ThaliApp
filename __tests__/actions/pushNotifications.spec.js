import * as actions from '../../app/actions/pushNotifications';

describe('push notifications actions', () => {
  it('should expose the push notifications actions', () => {
    expect(actions.REGISTER).toEqual('PUSH_NOTIFICATIONS_REGISTER');
    expect(actions.INVALIDATE).toEqual('PUSH_NOTIFICATIONS_INVALIDATE');
  });

  it('should create an action to register the push token', () => {
    expect(actions.register()).toMatchSnapshot();
  });

  it('should create an action to invalidate the push token', () => {
    expect(actions.invalidate()).toMatchSnapshot();
  });
});