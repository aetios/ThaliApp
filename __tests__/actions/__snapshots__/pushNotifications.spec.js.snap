// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`push notifications actions should create an action to invalidate the push token 1`] = `
Object {
  "type": "PUSH_NOTIFICATIONS_INVALIDATE",
}
`;

exports[`push notifications actions should create an action to register the push token 1`] = `
Object {
  "categories": null,
  "type": "PUSH_NOTIFICATIONS_REGISTER",
}
`;

exports[`push notifications actions should create an action to register the push tokens and update push categories 1`] = `
Object {
  "categories": Array [
    "category1",
    "category2",
  ],
  "type": "PUSH_NOTIFICATIONS_REGISTER",
}
`;
